app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,65 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, :valid_user, :check_expiration, only: [:edit, :update]
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      handle_create @user
+    else
+      flash.now[:danger] = t(".notfound")
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t("cant_empty")
+      render :edit
+    elsif @user.update user_params
+      handle_update @user
+    else
+      render :edit
+    end
+  end
+
+  private
+  def user_params
+    params.require(:user).permit(:password, :password_confirmation)
+  end
+
+  def get_user
+    return if @user = User.find_by(email: params[:email])
+    flash[:danger] = t(".user_not_exist")
+  end
+
+  def valid_user
+    return if @user&.activated? && @user&.authenticated?(:reset, params[:id])
+
+    redirect_to root_url
+  end
+
+  def check_expiration
+    return unless @user.password_reset_expired?
+
+    flash[:danger] = t(".password_expried")
+    redirect_to new_password_reset_url
```

Comment:
```suggestion
    redirect_to new_password_reset_path
```
trong các controller thì dùng path thay vì url e nhé